bindata/cloud-network-config-controller/managed/controller.yaml

```diff
@@ -184,6 +182,14 @@ spec:
 {{ if .NO_PROXY }}
         - name: "NO_PROXY"
           value: "{{ .NO_PROXY}}"
+{{ end }}
+{{- if not (eq .AroHcpSecretProviderClass "")}}
```

Comment:
In your CNCC [PR](https://github.com/openshift/cloud-network-config-controller/pull/156/files#diff-d3dbddd1556958367344cf05b435a98ffdc18ec9c060843fa1a5f4582d22f3b7R594) you are using `ARO_HCP_MI_CLIENT_ID` to determine whether to use Managed Identity or not. Can you use the same condition here?